src/Http/Http.Abstractions/src/RouteHandlerContext.cs

```diff
@@ -16,10 +16,12 @@ public sealed class RouteHandlerContext
     /// </summary>
     /// <param name="methodInfo">The <see cref="MethodInfo"/> associated with the route handler of the current request.</param>
     /// <param name="endpointMetadata">The <see cref="EndpointMetadataCollection"/> associated with the endpoint the filter is targeting.</param>
-    public RouteHandlerContext(MethodInfo methodInfo, EndpointMetadataCollection endpointMetadata)
+    /// <param name="applicationServices">The <see cref="IServiceProvider"/> instance used to access the application services.</param>
+    public RouteHandlerContext(MethodInfo methodInfo, EndpointMetadataCollection endpointMetadata, IServiceProvider? applicationServices)
```

Comment:
Why is it nullable?

Reply:
Because it's nullable on `RequestDelegateFactoryOptions` which is where it's set from. Same thing we did for `EndpointMetadataContext` and `EndpointParameterMetadataContext`.